pyasdf/tags/transform/selector.py

```diff
@@ -0,0 +1,77 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import, division, unicode_literals, print_function
+
+import numpy as np
+from numpy.testing import assert_array_equal
+
+
+from gwcs import selector
```

Comment:
This creates a dependency of pyasdf of gwcs, which we probably don't want to do.

There are some ways out here (and I'm Cc'ing @perrygreenfield as he might have some thoughts):

1) Move this type to the `wcs` module in pyasdf, so at least only `wcs` stuff, and not core `transform` stuff in ASDF will depend on `gwcs`.

2) Put the region-handling stuff in astropy.

3) Put the region-handling stuff (i.e. the actual code that handles regions, not just the reading to/from ASDF files) in pyasdf.  That's not as strange as it sounds if these classes exist only to handle ASDF WCS -- and if they are more general, (2) is probably the better choice.

4) Put this ASDF type and schema in the gwcs project, in which case this becomes a test bed for "third-party ad hoc" types in ASDF.

But as it stands, I don't think we want to grow a dependency on gwcs here.  I'm ultimately trying to get the astropy dependency out of pyasdf anyway (except to handle the transform stuff), since it's supposed to be non-astro-specific ultimately.  That should be more doable once some more features are added to astropy-helpers.